core/src/subgraph/runner.rs

```diff
@@ -217,16 +201,15 @@ where
         // very contrived subgraph would be able to observe this.
         while block_state.has_created_data_sources() {
             // Instantiate dynamic data sources, removing them from the block state.
-            let (data_sources, runtime_hosts) = self.create_dynamic_data_sources(
-                logger.clone(),
-                host_metrics.clone(),
-                block_state.drain_created_data_sources(),
-            )?;
+            let (data_sources, runtime_hosts) =
+                self.create_dynamic_data_sources(block_state.drain_created_data_sources())?;
```

Comment:
I wonder if it would be clearer to pass a &mut blockstate here and then let that function deal with the draining

Reply:
Hmmm, I kind of like the [current definition](https://github.com/graphprotocol/graph-node/blob/master/graph/src/components/subgraph/instance.rs#L69) that does an assert first.

Not that it couldn't be improved. Well I'll leave this one for out of scope of this PR, thanks for the review!!